ldms/src/contrib/sampler/tutorial/Makefile.am

```diff
@@ -0,0 +1,29 @@
+SUBDIRS =
+pkglib_LTLIBRARIES =
+lib_LTLIBRARIES =
+bin_SCRIPTS=
+dist_man7_MANS =
+dist_man1_MANS =
+EXTRA_DIST =
+
+CORE = ../../../core
+SAMPLER = ../../../sampler
+LDMSD = ../../../ldmsd
+BASE_SRC= $(srcdir)/$(SAMPLER)/sampler_base.c $(srcdir)/$(SAMPLER)/sampler_base.h
+
+AM_CFLAGS = -I$(srcdir)/$(CORE) -I$(top_srcdir) -I../../.. -I$(srcdir)/$(SAMPLER) @OVIS_LIB_INCDIR_FLAG@ \
+            -I$(srcdir)/$(LDMSD)
+
+AM_LDFLAGS = @OVIS_LIB_LIB64DIR_FLAG@ @OVIS_LIB_LIBDIR_FLAG@
+
+COMMON_LIBADD = $(CORE)/libldms.la @LDFLAGS_GETTIME@ -lovis_util -lcoll
+
+
+if ENABLE_TUTORIAL_SAMPLER
+libtutorial_sampler_la_SOURCES = tutorial_sampler.c $(BASE_SRC)
+libtutorial_sampler_la_LIBADD = $(COMMON_LIBADD)
+libtutorial_sampler_la_CFLAGS = $(AM_CFLAGS) 
+pkglib_LTLIBRARIES += libtutorial_sampler.la
+dist_man7_MANS += Plugin_tutorial_sampler.man
+
+endif
```

Comment:
We will want an "EXTRA_DIST = Plugin_tutorial_sampler.man" here, after the "endif".  This is an unfortunate workaround for a known bug in automake.  For things like _SOURCES, automake is smart enough to add the files to EXTRA_DIST automatically.  Sadly it was forgotten to do the same for _MANS.  I don't believe that even the dist_ prefix helps when the _MANS line appears inside of the conditional.

Reply:
@morrone I don't understand why this would go after the endif. If the "if" isn't met, there wouldn't be this additional file.

Reply:
We want all source files to appear in the "make dist" tarball regardless of whether the part was selected for creation at build and install time.  For instance, even though the _SOURCES right above this are inside the "if", those files _are_ included in the "make dist" tarball even when the "if" is not met.  autoconf should be doing the same with _MANS, but it has a bug and we need to help it.  EXTRA_DIST will accomplish that.

Reply:
Since its outside of the endif, if the user specifies a file that does not exist (e.g., EXTRA_DIST = FOOBAR.man), the make dist will fail. Are we ok with this? How much are we requiring in contrib plugins vs main line plugins? config is sane, but we (main line developers) aren't guaranteeing the plugin will actually work? @morrone @baallan @tom95858 @narategithub @valleydlr @nichamon